includes/enqueue-assets/class-enqueue-assets.php

```diff
@@ -50,5 +51,12 @@ public static function enqueue_styles_nonblocking() : void {
 			wp_enqueue_style( 'cata-nonblocking', get_template_directory_uri() . '/assets/dist/css/nonblocking.css', array(), wp_get_theme( 'cata' )->get( 'Version' ), 'screen' );
 		}
 
+		/**
+		 * Enqueue Scripts
+		 */
+		public static function enqueue_scripts() : void {
+			wp_enqueue_script( 'cata-module-app', get_stylesheet_directory_uri() . '/assets/dist/js/app.js', array(), wp_get_theme( 'cata' )->get( 'Version' ), true );
```

Comment:
This has `-module` in the slug, but that's not doing anything in the parent theme yet. In the child themes we sometimes use `themename-module` as a way to know when we should add `type="module"` as an attribute on the `<script>` tag.

I think you could add a filter on `script_loader_tag` that looks for `cata-module` and replaces `href=` with `type="module" href=`

That's how we can best handle adding the type attribute without messing up any other attributes.

The child theme rewrites the whole `<script>` element which has caused trouble once or twice.